keras/backend/tensorflow/optimizer.py

```diff
@@ -19,6 +19,23 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self._distribution_strategy = tf.distribute.get_strategy()
 
+    def build(self, variables):
+        super().build(variables)
+        if hasattr(self, "_ema_vars_initialized"):
+            # We must use `backend.Variable` in
+            # `self._update_model_variables_moving_average` to correctly
+            # initialize moving averages with the same value as
+            # `trainable_variables` during the first `self.apply`.
```

Comment:
If it is only used once, can it be a Python bool?

Reply:
I not sure why, but it failed when we used python boolean in the compiled function (graph mode as well).